### PR TITLE
ci: limit concurrent deploys from master

### PR DESCRIPTION
**Pull request by stoksc**
## Commentary (optional)
Maybe we can turn on 3rd party orbs for this - it's the most recent recommended solution to this https://discuss.circleci.com/t/sequentially-running-workflows/23288/2.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: .circleci/config.yml
Comment (dzhu) [inline .circleci/config.yml:1263]:
```suggestion
          # Basically wait forever -- we would prefer not to fail deploys and
          # we'll likely never be this backed up.
```

**Review by dzhu — APPROVED** [review]
Cool! I was pondering how we might do this ourselves; good thing a solution exists already. Maybe wait until after the release to land in case something goes awry.

Also, I think you may need to do something with setting up an API key before this will work, according to the [website](https://eddiewebb.github.io/circleci-queue/)?

**Comment by stoksc** [comment]
> Also, I think you may need to do something with setting up an API key before this will work, according to the website

Yeah I think we may need to do _something_.

**Comment by stoksc** [comment]
OK @dzhu, I added the API key with the default name to the Determined project so I think this should "just work".

**Comment by stoksc** [comment]
I'll land once release it cut and done.

**Pull request merged**
